packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js

```diff
@@ -356,64 +356,70 @@ const HomePage = () => {
       dataToDelete.find(itemToDelete => item.id.toString() === itemToDelete.id.toString())
     ) && hasSomeCheckboxSelected;
 
-  if (isLoading) {
-    return <LoadingIndicatorPage />;
-  }
-
   const filters = generateFiltersFromSearch(search);
   const hasFilters = !isEmpty(filters);
   const hasSearch = !isEmpty(searchValue);
   const areResultsEmptyWithSearchOrFilters = isEmpty(data) && (hasSearch || hasFilters);
 
   return (
     <Container>
-      <Header {...headerProps} />
-      <HeaderSearch
-        label={pluginName}
-        onChange={handleChangeSearchValue}
-        onClear={handleClearSearch}
-        placeholder={formatMessage({ id: getTrad('search.placeholder') })}
-        name="_q"
-        value={searchValue}
-      />
-      <ControlsWrapper>
-        <SelectAll
-          onChange={handleSelectAll}
-          checked={areAllCheckboxesSelected}
-          someChecked={hasSomeCheckboxSelected && !areAllCheckboxesSelected}
-        />
-        <Padded right />
-        <SortPicker
-          onChange={handleChangeParams}
-          value={query.get('_sort') || `${updated_at}:DESC`}
-        />
-        <Padded right />
-        <Filters onChange={handleChangeParams} filters={filters} onClick={handleDeleteFilter} />
-      </ControlsWrapper>
-      {dataCount > 0 && !areResultsEmptyWithSearchOrFilters ? (
+      <Header {...headerProps} isLoading={isLoading} />
+      {isLoading && (
+        <>
+          <Padded top bottom size="lg" />
+          <LoadingIndicator />
+        </>
+      )}
+      {!isLoading && (
```

Comment:
You don't need a second condition here. You can use only one
```js
{isLoading ? (
<>
     ...    
</>
) : (
<>
     ...
</>
)}
```

Reply:
Also, this homepage component is getting longer and longer so do not hesitate to break it up into multiple components.
For example: 
```js
{isLoading ? (
     <HomePageLoader /> 
) : (
     <HomePageContent /> // Pass the needed props and manage the state of this component inside this component
)}
```

Reply:
This will avoid useless renders of the loader